jobserver/forms.py

```diff
@@ -113,13 +113,18 @@ def __init__(self, project, repos_with_branches, *args, **kwargs):
 
         # has there been a repo selected already?
         if "data" in kwargs and "repo" in kwargs["data"]:
-            repo = next(  # pragma: no branch
-                (
-                    r
-                    for r in self.repos_with_branches
-                    if r["url"] == kwargs["data"]["repo"]
-                ),
-            )
+            try:
+                repo = next(
+                    (
+                        r
+                        for r in self.repos_with_branches
+                        if r["url"] == kwargs["data"]["repo"]
+                    ),
+                )
+            except StopIteration:
+                raise forms.ValidationError(
+                    "Unknown repo, please reload the page and try again"
+                )
```

Comment:
thought: This comes from `get_github_api().get_repos_with_branches(gh_org)` in `WorkspaceCreate`. That in turn pulls repositories from the OpenSAFELY researchers team.

We *always* expect that will contain repositories with branches.

I guess it's _possible_ that GitHub's API is working and gives a 200 response, but is broken and gives no repositories.

Also, I copied the message from elsewhere in the code, but maybe the error should be:

> No matching repos found, please reload the page and try again

Reply:
Also, I copied the message from elsewhere in the code, but maybe the error should be:

> No matching repos found, please reload the page and try again

Reply:
I updated the error message.